iree/compiler/Dialect/HAL/Target/LLVM/AOT/internal/UnixLinkerTool.cpp

```diff
@@ -31,7 +32,43 @@ class UnixLinkerTool : public LinkerTool {
 
   std::string getToolPath() const override {
     auto toolPath = LinkerTool::getToolPath();
-    return toolPath.empty() ? "ld.lld" : toolPath;
+    if (!toolPath.empty()) return toolPath;
+    if (targetTriple.isAndroid()) {
+      char *androidNDKPath = std::getenv("ANDROID_NDK");
+      if (!androidNDKPath) return toolPath;
+// TODO(ataei, benvanik): Windows cross-linking android NDK support.
```

Comment:
Wouldn't these TODOs go in WindowsLinkerTool.cpp?

Around here: https://github.com/google/iree/blob/489046b3de7262f62473fa2a074b76f8eae8ec93/iree/compiler/Dialect/HAL/Target/LLVM/AOT/internal/WindowsLinkerTool.cpp#L176-L196

or here: https://github.com/google/iree/blob/489046b3de7262f62473fa2a074b76f8eae8ec93/iree/compiler/Dialect/HAL/Target/LLVM/AOT/internal/WindowsLinkerTool.cpp#L32-L35

depending on what needs to change.

Maybe just the Windows TODOs from this PR?

Reply:
The windows linker tool is for Windows targets, the TODO here for when compiling for Andorid target on windows host

Reply:
Aaaah, I see. Curious if that will work as set up :P 

https://github.com/google/iree/blob/489046b3de7262f62473fa2a074b76f8eae8ec93/iree/compiler/Dialect/HAL/Target/LLVM/AOT/internal/LinkerTools.cpp#L34-L37

Reply:
hmmm, shouldn't have `isWindowsMSVCEnvironment` if we must have it here then the TODO I have needs to move to `createWindowsLinkerTool` but I think we can keep linker to be target specific and do similar thing (e.g build for windows on linux ?)